app-backend/datamodel/src/main/scala/Scope.scala

```diff
@@ -0,0 +1,308 @@
+          action.repr
+        }).mkString(";"))
+    }
+  }
+
+  implicit val decScope: Decoder[Scope] = new Decoder[Scope] {
+    def apply(c: HCursor): Decoder.Result[Scope] = c.value.asString match {
+      case Some("uploader")             => Right(Scopes.Uploader)
+      case Some("analyses:crud")        => Right(Scopes.AnalysesCRUD)
+      case Some("analyses:multiplayer") => Right(Scopes.AnalysesMultiPlayer)
+      case Some("datasources:crud")     => Right(Scopes.DatasourcesCRUD)
+      case Some("organizations:userAdmin") =>
+        Right(Scopes.OrganizationsUserAdmin)
+      case Some("projects:exportFullAccess") => Right(Scopes.ProjectExport)
+      case Some("projects:fullAccess")       => Right(Scopes.ProjectsFullAccess)
+      case Some("organizations:admin") =>
+        Right(Scopes.RasterFoundryOrganizationAdmin)
+      case Some("teams:admin")           => Right(Scopes.RasterFoundryTeamAdmin)
+      case Some("scenes:crud")           => Right(Scopes.ScenesCRUD)
+      case Some("scenes:multiplayer")    => Right(Scopes.ScenesMultiPlayer)
+      case Some("shapes:fullAccess")     => Right(Scopes.ShapesFullAccess)
+      case Some("teams:edit")            => Right(Scopes.TeamsEdit)
+      case Some("templates:crud")        => Right(Scopes.TemplatesCRUD)
+      case Some("templates:multiplayer") => Right(Scopes.TemplatesMultiPlayer)
+      case Some("uploads:crud")          => Right(Scopes.UploadsCRUD)
+      case Some(s) =>
+        SimpleScope.fromEithers((s.split(";").toList match {
+          case List("") => List.empty
+          case actions =>
+            actions.map(
+              act =>
+                decode[Action](s""""$act"""")
+                  .leftMap(err => DecodingFailure(err.getMessage, Nil))
+            )
+        }))
+      case _ => Left(DecodingFailure("Scope", c.history))
+    }
+  }
+}
+
+object Scopes {
+
+  private def makeCRUDActions(domain: String): Set[Action] =
+    Set("read", "create", "update", "delete") map { s =>
+      makeAction(domain, s)
+    }
+
+  private def makeAction(
+      domain: String,
+      action: String,
+      limit: Option[Long] = None
+  ): Action =
+    Action(domain, action, limit)
+
+  case object NoAccess extends SimpleScope(Set.empty)
+
+  case object Uploader
+      extends SimpleScope(
+        Set("read", "create", "delete") map { makeAction("uploads", _) }
+      )
+
+  case object UploadsCRUD
+      extends ComplexScope(
+        Set(
+          Uploader,
+          new SimpleScope(Set(makeAction("uploads", "delete")))
+        )
+      )
+
+  case object ScenesCRUD extends SimpleScope(makeCRUDActions("scenes"))
+
+  case object ScenesMultiPlayer
+      extends SimpleScope(Set(makeAction("scenes", "share")))
+
+  case object ScenesFullAccess
+      extends ComplexScope(Set(ScenesCRUD, ScenesMultiPlayer))
+
+  case object ProjectsCRUD extends SimpleScope(makeCRUDActions("projects"))
+
+  case object ProjectExport
```

Comment:
These are all `case object`s instead of `val`s because it lets us do pattern-matching to enable complex scopes to become magic strings